### PR TITLE
feat: Add patch endpoint to update contact details

### PR DESCRIPTION
**Pull request by Judge40**
TISNEW-5154

### CODE REVIEW

File: src/test/java/uk/nhs/hee/trainee/details/api/ContactDetailsResourceTest.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
+package uk.nhs.hee.trainee.details.api;
```
Comment (JosephKelly) [inline src/test/java/uk/nhs/hee/trainee/details/api/ContactDetailsResourceTest.java:1]: License

File: src/test/java/uk/nhs/hee/trainee/details/service/impl/PersonalDetailsServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
+package uk.nhs.hee.trainee.details.service.impl;
```
Comment (JosephKelly) [inline src/test/java/uk/nhs/hee/trainee/details/service/impl/PersonalDetailsServiceImplTest.java:1]: License here too

**Review by JosephKelly — COMMENTED** [review]
Looks like a good trade off between pure RESTfulness & what we need.  Couple of reminders before merging.  Nice choice of id to test.

File: src/test/java/uk/nhs/hee/trainee/details/api/ContactDetailsResourceTest.java (hunk 1 above)
Comment (Judge40) [reply src/test/java/uk/nhs/hee/trainee/details/api/ContactDetailsResourceTest.java:1]: Ugh, IntelliJ doesn't add it when creating test classes via class context menu it seems. I'll fix.

**Review by jayanta2018 — APPROVED** [review]

**Review by JosephKelly — APPROVED** [review]

**Pull request merged**